v2/blockstore/example_test.go

```diff
@@ -0,0 +1,146 @@
+package blockstore_test
+
+import (
+	"context"
+	"fmt"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-merkledag"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/blockstore"
+)
+
+// ExampleOpenReadOnly opens a read-only blockstore from a CARv1 file, and prints its root CIDs
+// along with CID mapping to raw data size of blocks for first five sections in the CAR file.
+func ExampleOpenReadOnly() {
+	// Open a new ReadOnly blockstore from a CARv1 file.
+	// Note, `OpenReadOnly` accepts bot CARv1 and CARv2 formats and transparently generate index
+	// in the background if necessary.
+	// This instance sets ZeroLengthSectionAsEOF option to treat zero sized sections in file as EOF.
+	robs, err := blockstore.OpenReadOnly("../testdata/sample-v1.car", carv2.ZeroLengthSectionAsEOF)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := robs.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// Print root CIDs.
+	roots, err := robs.Roots()
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Contains %v root CID(s):\n", len(roots))
+	for _, r := range roots {
+		fmt.Printf("\t%v\n", r)
+	}
+
+	// Print the raw data size for the first 5 CIDs in the CAR file.
+	keysChan, err := robs.AllKeysChan(context.Background())
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("List of first 5 CIDs and their raw data size:")
+	var i int
+	for k := range keysChan {
+		if i > 4 {
```

Comment:
if you want to print 5, I think it's then easier to do `i >= 5`, to use the same number everywhere. you might as well make that a const/var and use it in the print above